src/network/networkd-dhcp4.c

```diff
@@ -138,6 +139,93 @@ static bool link_prefixroute(Link *link) {
+        if (r < 0)
+                return log_link_warning_errno(link, r, "DHCP error: no address: %m");
+
+        r = sd_dhcp_lease_get_netmask(link->dhcp_lease, &netmask);
+        if (r < 0)
+                return log_link_warning_errno(link, r, "DHCP error: no netmask: %m");
+
+        prefixlen = in4_addr_netmask_to_prefixlen(&netmask);
+
+        (void) sd_dhcp_lease_get_router(link->dhcp_lease, &router);
+
+        if (DEBUG_LOGGING) {
+                _cleanup_free_ char *pretty = NULL;
+
+                (void) in_addr_to_string(AF_INET, &address, &pretty);
+                log_link_debug(link, "DHCP: Configuring RPDB rule from DHCPv4 address %s", strna(pretty));
+        }
+
+        HASHMAP_FOREACH(rule, link->network->rules_by_section) {
+                if (!IN_SET(rule->to_type, DHCP4_ADDRESS_TYPE_DHCP4_IP, DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY) &&
+                    !IN_SET(rule->from_type, DHCP4_ADDRESS_TYPE_DHCP4_IP, DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY))
+                        continue;
+
+                if ((rule->to_type == DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY || rule->from_type == DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY) && !router) {
+                        log_link_debug(link, "DHCP error: Failed to configure RPDB rule from DHCPv4 gateway");
+                        continue;
+                }
+
+                if (rule->from_type == DHCP4_ADDRESS_TYPE_DHCP4_IP) {
+                        rule->from.in = address.in;
+                        rule->from_prefixlen = prefixlen;
+                }
+
+                if (rule->to_type == DHCP4_ADDRESS_TYPE_DHCP4_IP) {
+                        rule->to.in = address.in;
+                        rule->to_prefixlen = prefixlen;
+                }
+
+                if (rule->from_type == DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY) {
+                        rule->from.in = *router;
+                        rule->from_prefixlen = 32;
+                }
+
+                if (rule->to_type == DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY) {
+                        rule->to.in = *router;
+                        rule->to_prefixlen = 32;
+                }
+
+                r = routing_policy_rule_configure(rule, link);
+                if (r < 0)
+                        return log_link_warning_errno(link, r, "Could not set routing policy rule from DHCP4: %m");
+        }
+
+        return 0;
+}
+
+static void dhcp_remove_routing_policy_rules(Link *link) {
+        RoutingPolicyRule *rule;
+        int r;
+
+        assert(link);
+        assert(link->network);
+        assert(link->manager);
+
+        HASHMAP_FOREACH(rule, link->network->rules_by_section) {
+                if (!IN_SET(rule->to_type, DHCP4_ADDRESS_TYPE_DHCP4_IP, DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY) &&
+                    !IN_SET(rule->from_type, DHCP4_ADDRESS_TYPE_DHCP4_IP, DHCP4_ADDRESS_TYPE_DHCP4_GATEWAY))
+                        continue;
+
+                r = routing_policy_rule_remove(rule, link->manager);
```

Comment:
Please do not reuse `rule` here, but re-assign `from` and/or `to`, as the .network file may be used by multiple interfaces. In that case, the `from` and `to` addresses of this `rule` may be assigned to another interface.

Reply:
yes but if some other parameters are configured by the user like `Table=`, the rules wont's match . Hence a comparison is required with extraction.